### DISCUSSION TITLE
How far does a rebuild cascade

### DISCUSSION THREAD

**Discussion by derolf** [rrousselGit/riverpod#2535] [Q&A]
Assume I have 

* a Provider A
* a Provider B hat watches on A
* a ConsumerWidget W that watches B

Let’s assume A gets invalidated, but returns exactly the same value on the next build.

* Will B be invalidated at the same moment A was invalidated?
* Will B be rebuilt at the next frame even though the watched value is identical?
* Will W be having its state set dirty?
* Will W be rebuilt at the next frame?

**Comment by rrousselGit** [comment]
> Will B be invalidated at the same moment A was invalidated?

No, it will be invalidated when ref.listen(a) is invoked.
If the listener is invoked of course.

Depends on the provider for whether A will notify. If it's a FutureProvider/StreamProvider, a new value is almost always emitted (loading).
For a Provider, depends on the value's ==


From there, if a update, b will be invalidated and update within the same frame (if and only if it is watched)
